firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift

```diff
@@ -1908,10 +1912,15 @@ class BrowserViewController: UIViewController,
         case .estimatedProgress:
             guard tab === tabManager.selectedTab else { break }
             if let url = webView.url, !InternalURL.isValid(url: url) {
+                let progress = if let progress = change?[.newKey] as? Double {
+                    progress
+                } else {
+                    webView.estimatedProgress
+                }
```

Comment:
Why is this necessary? I think this change notify the progress bar even if the progress value hasn't changed?

Reply:
Discussed with Filippo, necessary since `onDownloadProgressUpdate` pass the estimatedProgress so we update the progress bar